src/card_operations/card_sign.h

```diff
@@ -0,0 +1,98 @@
+ * @copyright Copyright (c) 2023 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/"
+ * target=_blank>https://mitcc.org/</a>
+ */
+#ifndef CARD_SIGN_H
+#define CARD_SIGN_H
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "card_operations.h"
+#include "stdbool.h"
+#include "stdint.h"
+
+/*****************************************************************************
+ * MACROS AND DEFINES
+ *****************************************************************************/
+#define FAMILY_ID_SIZE 4
+#define INIT_SIGN_SERIAL_CONFIG(acceptable_cards,                              \
+                                family_id,                                     \
+                                skip_card_removal,                             \
+                                heading,                                       \
+                                message,                                       \
+                                serial_buffer)                                 \
+                                                                               \
+  {                                                                            \
+    acceptable_cards, family_id, CARD_SIGN_SERIAL, skip_card_removal, heading, \
+        message, serial_buffer, 0, {                                           \
+      0                                                                        \
+    }                                                                          \
+  }
+
+#define INIT_SIGN_CUSTOM_CONFIG(acceptable_cards,                              \
+                                family_id,                                     \
+                                skip_card_removal,                             \
+                                heading,                                       \
+                                message,                                       \
+                                challenge_buffer,                              \
+                                challenge_size)                                \
+                                                                               \
+  {                                                                            \
+    acceptable_cards, family_id, CARD_SIGN_CUSTOM, skip_card_removal, heading, \
+        message, challenge_buffer, challenge_size, {                           \
+      0                                                                        \
+    }                                                                          \
+  }
+
+/*****************************************************************************
+ * TYPEDEFS
+ *****************************************************************************/
+typedef enum card_sign_type {
+  CARD_SIGN_SERIAL = 1,
+  CARD_SIGN_CUSTOM,
```

Comment:
this distinction is not necessary

Reply:
Both cases are different in their data source.

In both cases, a data buffer is required but in the serial case, the data is set internally from family id and card number.

Reply:
Will add an explanation in enum declaration as well.

Reply:
okay. this makes sense